src/bin/inx-chronicle/api/stardust/mod.rs

```diff
@@ -6,10 +6,21 @@ use super::router::Router;
 pub mod core;
 pub mod explorer;
 pub mod indexer;
+#[cfg(feature = "poi")]
+pub mod poi;
 
+#[allow(clippy::let_and_return)]
 pub fn routes() -> Router {
-    Router::new()
-        .nest("/explorer/v2", explorer::routes())
+    #[allow(unused_mut)]
+    let mut router = Router::new()
         .nest("/core/v2", core::routes())
-        .nest("/indexer/v1", indexer::routes())
+        .nest("/explorer/v2", explorer::routes())
+        .nest("/indexer/v1", indexer::routes());
+
+    #[cfg(feature = "poi")]
+    {
+        router = router.nest("/poi/v1", poi::routes());
+    }
+
+    router
```

Comment:
Well, there's a reason we don't use it like this. Because the macro doesn't remove branches at compile time, this will fail to compile without the `poi` feature because the `poi` module won't exist.

Reply:
exactly....I wanted to use the `cfg!` macro several times now (also in Bee) only to find that you can make use of it relatively rarely.